### ISSUE TITLE
Parser blow up for long linear combinations

### ISSUE THREAD

**Issue by Schaeff** [Zokrates/ZoKrates#60]
Parsing this simple program:
```
def main():
	a = 1+2*3+3*4+4*5+5*6+6*7+7*8+8*9+9*10+10*11+11*12+12*13+13*14
	return 1
```

takes a surprisingly long time, which I identified to be due to a problem in the parser which jumps backwards a lot

**Comment by Schaeff** [comment]
This will be fixed by #306

**Comment by Schaeff** [comment]
Fixed!
